File: src/main/java/com/ilCary/CarUnit/models/Address.java
package com.ilCary.CarUnit.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "addresses")
@Data
@NoArgsConstructor
@Builder
@AllArgsConstructor
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String street; //via

    private String streetNum;//civico


    @ManyToOne
    private Comune municipality; //comune

}

File: CarUnit_fe/CarUnit/src/app/models/Address.ts
import { Comune } from "./Comune";

export class Address {
  id?: number;
  street: string;
  streetNum: string;
  municipality: Comune;

  constructor(street: string, streetNum: string,municipality:Comune) {
    this.street = street;
    this.streetNum = streetNum;
    this.municipality = municipality;
  }

}
